pkg/k8s/cnp.go

```diff
@@ -306,9 +363,67 @@ func (c *CNPStatusUpdateContext) update(cnp *types.SlimCNP, enforcing, ok bool,
 
 	cnpns = createCNPNodeStatus(enforcing, ok, cnpError, rev, annotations)
 
-	ns := k8sUtils.ExtractNamespace(&cnp.ObjectMeta)
+	switch m.typ {
+	case K8sAPIServer:
+		ns := k8sUtils.ExtractNamespace(&cnp.ObjectMeta)
+		return updateStatusesByCapabilities(c.CiliumNPClient, capabilities, cnp, ns, cnp.GetName(), map[string]cilium_v2.CiliumNetworkPolicyNodeStatus{c.NodeName: cnpns})
+	case KVStore:
+		cnpWithMeta := CNPNSWithMeta{
```

Comment:
`CiliumNetworkPolicyNodeStatus` is the only value we need. All of the other fields (name, namespace, uid, nodename) are part of the key name as formatted in line 383

Reply:
I found it cleaner to be able to unmarshal and access the keys instead of having to do string manipulation - is the motivation behind this to avoid duplication of data stored in the key value store?

Reply:
yes

Reply:
Actually, I remember why I had to do it this way - I need to be able to implement `GetKeyName` to implement the `Key` interface so that I can store key-value pairs in the `SharedStore`. So, I need to encapsulate all of the information which is needed for the key corresponding to the value, within the value itself - the UUID/Namespace/Name/NodeName.